File: scripts/inspect_rodrigo.py

import psycopg2
from psycopg2.extras import RealDictCursor

db_url = "postgresql://postgres:dWldGAyqtVAsQvMYvTIqBKXTkHedCLAS@shortline.proxy.rlwy.net:26559/railway"

def inspect_supplier():
    conn = psycopg2.connect(db_url)
    cur = conn.cursor(cursor_factory=RealDictCursor)
    
    # 1. Find supplier ID
    cur.execute("SELECT id, nome FROM fornecedores WHERE nome ILIKE %s", ('%RODRIGO GASPAR%',))
    suppliers = cur.fetchall()
    
    if not suppliers:
        print("Supplier 'RODRIGO GASPAR' not found.")
        return
    
    for supplier in suppliers:
        s_id = supplier['id']
        name = supplier['nome']
        print(f"Checking Supplier: {name} (ID: {s_id})")
        
        # Counts in various tables
        tables = [
            ('solicitacoes', 'fornecedor_id'),
            ('ordens_compra', 'fornecedor_id'),
            ('lotes', 'fornecedor_id'),
            ('fornecedor_tipo_lote_precos', 'fornecedor_id'),
            ('fornecedor_tabela_precos', 'fornecedor_id'),
            ('fornecedor_tipo_lote', 'fornecedor_id'),
            ('fornecedor_classificacao_estrela', 'fornecedor_id'),
            ('fornecedor_funcionario_atribuicao', 'fornecedor_id')
        ]
        
        for table, col in tables:
            cur.execute(f"SELECT COUNT(*) FROM {table} WHERE {col} = %s", (s_id,))
            count = cur.fetchone()['count']
            print(f"  Table {table}: {count} records")

        # Indirectly related (via solicitacoes)
        cur.execute("SELECT id FROM solicitacoes WHERE fornecedor_id = %s", (s_id,))
        sol_ids = [r['id'] for r in cur.fetchall()]
        if sol_ids:
            cur.execute("SELECT COUNT(*) FROM itens_solicitacao WHERE solicitacao_id IN %s", (tuple(sol_ids),))
            count = cur.fetchone()['count']
            print(f"  Table itens_solicitacao: {count} records")

        # Indirectly related (via ordens_compra)
        cur.execute("SELECT id FROM ordens_compra WHERE fornecedor_id = %s", (s_id,))
        oc_ids = [r['id'] for r in cur.fetchall()]
        if oc_ids:
            cur.execute("SELECT COUNT(*) FROM ordens_servico WHERE oc_id IN %s", (tuple(oc_ids),))
            count = cur.fetchone()['count']
            print(f"  Table ordens_servico: {count} records")
            
            # Via OS
            cur.execute("SELECT id FROM ordens_servico WHERE oc_id IN %s", (tuple(oc_ids),))
            os_ids = [r['id'] for r in cur.fetchall()]
            if os_ids:
                for t in ['gps_logs', 'rotas_operacionais', 'conferencias_recebimento']:
                    cur.execute(f"SELECT COUNT(*) FROM {t} WHERE os_id IN %s", (tuple(os_ids),))
                    count = cur.fetchone()['count']
                    print(f"    Table {t}: {count} records")

        # Indirectly related (via lotes)
        cur.execute("SELECT id FROM lotes WHERE fornecedor_id = %s", (s_id,))
        lote_ids = [r['id'] for r in cur.fetchall()]
        if lote_ids:
            for t in ['entradas_estoque', 'movimentacoes_estoque', 'lotes_separacao', 'inventario_contagens']:
                cur.execute(f"SELECT COUNT(*) FROM {t} WHERE lote_id IN %s", (tuple(lote_ids),))
                count = cur.fetchone()['count']
                print(f"  Table {t}: {count} records")
            
            # Via Lotes Separacao
            cur.execute("SELECT id FROM lotes_separacao WHERE lote_id IN %s", (tuple(lote_ids),))
            sep_ids = [r['id'] for r in cur.fetchall()]
            if sep_ids:
                cur.execute("SELECT COUNT(*) FROM residuos WHERE separacao_id IN %s", (tuple(sep_ids),))
                count = cur.fetchone()['count']
                print(f"    Table residuos: {count} records")

    cur.close()
    conn.close()

if __name__ == "__main__":
    inspect_supplier()

File: scripts/inspect_rodrigo.js

const { Client } = require('pg');

const connectionString = "postgresql://postgres:dWldGAyqtVAsQvMYvTIqBKXTkHedCLAS@shortline.proxy.rlwy.net:26559/railway";

async function inspect() {
    const client = new Client({ connectionString });
    await client.connect();

    try {
        // 1. Find supplier
        const resFornecedor = await client.query("SELECT id, nome FROM fornecedores WHERE nome ILIKE '%RODRIGO GASPAR%'");
        if (resFornecedor.rows.length === 0) {
            console.log("Fornecedor 'RODRIGO GASPAR' não encontrado.");
            return;
        }

        for (const fornecedor of resFornecedor.rows) {
            const s_id = fornecedor.id;
            console.log(`\nAnalisando Fornecedor: ${fornecedor.nome} (ID: ${s_id})`);

            // Tables with direct supplier_id
            const directTables = [
                ['solicitacoes', 'fornecedor_id'],
                ['ordens_compra', 'fornecedor_id'],
                ['lotes', 'fornecedor_id'],
                ['fornecedor_tipo_lote_precos', 'fornecedor_id'],
                ['fornecedor_tabela_precos', 'fornecedor_id'],
                ['fornecedor_tipo_lote', 'fornecedor_id'],
                ['fornecedor_classificacao_estrela', 'fornecedor_id'],
                ['fornecedor_funcionario_atribuicao', 'fornecedor_id']
            ];

            for (const [table, col] of directTables) {
                const countRes = await client.query(`SELECT COUNT(*) FROM ${table} WHERE ${col} = $1`, [s_id]);
                console.log(`  Tabela ${table}: ${countRes.rows[0].count} registros`);
            }

            // Indirect: via solicitacoes -> itens_solicitacao
            const solRes = await client.query("SELECT id FROM solicitacoes WHERE fornecedor_id = $1", [s_id]);
            const solIds = solRes.rows.map(r => r.id);
            if (solIds.length > 0) {
                const itemSolRes = await client.query(`SELECT COUNT(*) FROM itens_solicitacao WHERE solicitacao_id = ANY($1)`, [solIds]);
                console.log(`  Tabela itens_solicitacao: ${itemSolRes.rows[0].count} registros`);
            }

            // Indirect: via ordens_compra -> ordens_servico -> logs/conferencias
            const ocRes = await client.query("SELECT id FROM ordens_compra WHERE fornecedor_id = $1", [s_id]);
            const ocIds = ocRes.rows.map(r => r.id);
            if (ocIds.length > 0) {
                const osRes = await client.query("SELECT id FROM ordens_servico WHERE oc_id = ANY($1)", [ocIds]);
                const osIds = osRes.rows.map(r => r.id);
                console.log(`  Tabela ordens_servico: ${osRes.rows.length} registros`);
                
                if (osIds.length > 0) {
                    const osSubTables = ['gps_logs', 'rotas_operacionais', 'conferencias_recebimento'];
                    for (const table of osSubTables) {
                        const subCount = await client.query(`SELECT COUNT(*) FROM ${table} WHERE os_id = ANY($1)`, [osIds]);
                        console.log(`    Tabela ${table}: ${subCount.rows[0].count} registros`);
                    }
                }
            }

            // Indirect: via lotes -> estoque/movimentacao/separacao
            const loteRes = await client.query("SELECT id FROM lotes WHERE fornecedor_id = $1", [s_id]);
            const loteIds = loteRes.rows.map(r => r.id);
            if (loteIds.length > 0) {
                const loteSubTables = ['entradas_estoque', 'movimentacoes_estoque', 'lotes_separacao', 'inventario_contagens'];
                for (const table of loteSubTables) {
                    const subCount = await client.query(`SELECT COUNT(*) FROM ${table} WHERE lote_id = ANY($1)`, [loteIds]);
                    console.log(`  Tabela ${table}: ${subCount.rows[0].count} registros`);
                }

                // via lotes_separacao -> residuos
                const sepRes = await client.query("SELECT id FROM lotes_separacao WHERE lote_id = ANY($1)", [loteIds]);
                const sepIds = sepRes.rows.map(r => r.id);
                if (sepIds.length > 0) {
                    const resCount = await client.query(`SELECT COUNT(*) FROM residuos WHERE separacao_id = ANY($1)`, [sepIds]);
                    console.log(`    Tabela residuos: ${resCount.rows[0].count} registros`);
                }
            }
        }

    } catch (err) {
        console.error(err);
    } finally {
        await client.end();
    }
}

inspect();
